src/aliceVision/image/io.cpp

```diff
@@ -82,6 +82,72 @@ EImageColorSpace getImageColorSpace(const std::string& imagePath)
     return EImageColorSpace_stringToEnum(colorSpace);
 }
 
+std::string getImageColorSpace(const OIIO::ImageSpec& oiioSpec, const std::string& defaultColorSpace = "", const std::string& imagePath = "")
+{
+    std::string colorSpaceFromFileName = "";
+    if (!imagePath.empty())
+    {
+        colorSpaceFromFileName = getGlobalColorConfigOCIO().getColorSpaceFromFilepath(imagePath);
+    }
+
+    std::map<std::string, std::string> mapColorSpaces;
+
+    for (const auto& m : oiioSpec.extra_attribs)
+    {
+        const std::string name = m.name().string();
+
+        if (name.find("oiio:ColorSpace") != name.npos)
+        {
+            mapColorSpaces.emplace("oiio:ColorSpace", m.get_string());
+        }
+        else if (name.find("AliceVision:ColorSpace") != name.npos)
+        {
+            mapColorSpaces.emplace("AliceVision:ColorSpace", m.get_string());
+        }
+        else if (name.find(":workPlateColourSpace") != name.npos)
+        {
+            mapColorSpaces.emplace("workPlateColourSpace", m.get_string());
+        }
+        else if (name.find(":originalPlateColourSpace") != name.npos)
+        {
+            mapColorSpaces.emplace("originalPlateColourSpace", m.get_string());
+        }
+    }
+
+    std::string colorSpace = defaultColorSpace;
+
+    if (mapColorSpaces.find("AliceVision:ColorSpace") != mapColorSpaces.end())
+    {
+        colorSpace = mapColorSpaces.at("AliceVision:ColorSpace");
+    }
+    else if (mapColorSpaces.find("workPlateColourSpace") != mapColorSpaces.end())
+    {
+        colorSpace = mapColorSpaces.at("workPlateColourSpace");
+    }
+    else if (!colorSpaceFromFileName.empty())
+    {
+        colorSpace = colorSpaceFromFileName;
+    }
+    else if (mapColorSpaces.find("oiio:ColorSpace") != mapColorSpaces.end())
+    {
+        colorSpace = mapColorSpaces.at("oiio:ColorSpace");
+    }
+    else if (mapColorSpaces.find("originalPlateColourSpace") != mapColorSpaces.end())
+    {
+        colorSpace = mapColorSpaces.at("originalPlateColourSpace");
+    }
+
+    ALICEVISION_LOG_TRACE("Detected image color space: " << colorSpace);
+
+    if (!EImageColorSpace_isSupportedOIIOstring(colorSpace))
+    {
+        colorSpace = defaultColorSpace;
+        ALICEVISION_LOG_WARNING("Detected color space " << colorSpace << " is not supported. Set image color space to " << defaultColorSpace);
+    }
```

Comment:
Wouldn't it be better to check whether a given color space is supported before adding it to `mapColorSpaces` (and then applying the priority order only on those valid color spaces)?

For now, if you have for example two keys, and the one with the highest priority has an invalid color space while the second key has a valid one, we use the default color space anyways. Is that the behaviour we want?

Reply:
Given priority to keys over color space existence seems to be more appropriate as soon as a warning is raised if a non existing color space is detected. The key "AliceVision:ColorSpace" can only contain an existing color space. The second one which is supposed to be a user dedicated key is supposed to contain an existing color space. The others keys and the filename option fallback solution,